slack_sdk/web/internal_utils.py

```diff
@@ -255,38 +255,39 @@ def _warn_if_text_or_attachment_fallback_is_missing(endpoint: str, kwargs: Dict[
     if skip_deprecation:
         return
 
-    # At this point, at a minimum, text argument is missing. Warn the user about this.
-    message = (
+    # if text argument is missing, Warn the user about this.
+    # However, do not warn if a fallback field exists for all attachments, since this can be substituted.
+    missing_text_message = (
         f"The top-level `text` argument is missing in the request payload for a {endpoint} call - "
         f"It's a best practice to always provide a `text` argument when posting a message. "
         f"The `text` argument is used in places where content cannot be rendered such as: "
         "system push notifications, assistive technology such as screen readers, etc."
     )
-    warnings.warn(message, UserWarning)
+
+    # https://api.slack.com/reference/messaging/attachments
+    # Check if the fallback field exists for all the attachments
+    # Not all attachments have a fallback property; warn about this too!
+    missing_fallback_message = (
+        f"Additionally, the attachment-level `fallback` argument is missing in the request payload for a {endpoint} call"
+        f" - To avoid this warning, it is recommended to always provide a top-level `text` argument when posting a"
+        f" message. Alternatively you can provide an attachment-level `fallback` argument, though this is now considered"
+        f" a legacy field (see https://api.slack.com/reference/messaging/attachments#legacy_fields for more details)."
```

Comment:
In these three lines of string, we don't have any string interpolation. Need to remove f prefix to resolve flake8 warnings.

```suggestion
        " - To avoid this warning, it is recommended to always provide a top-level `text` argument when posting a"
        " message. Alternatively you can provide an attachment-level `fallback` argument, though this is now considered"
        " a legacy field (see https://api.slack.com/reference/messaging/attachments#legacy_fields for more details)."
```